zerver/lib/bugdown/__init__.py

```diff
@@ -1163,11 +1145,15 @@ def run(self, root: Element) -> None:
                 continue
 
             if extracted_data:
-                if youtube is not None:
-                    title = self.youtube_title(extracted_data)
-                    if title is not None:
-                        found_url.family.child.text = title
-                    continue
+                if youtube_urls:
+                    # We use a fix sized thumbnail for every YouTube preview. So,
+                    # thumbnail resolution is always converted to `default.jpg`.
+                    self.resize_youtube_image(extracted_data)
+                    if self.markdown.url_embed_preview_enabled:
```

Comment:
I think we can remove the check for whether `url_embed_preview_enabled` is True here. This code for replacing the URL with the title was here, so as to not make a request for open graph data when the feature is not enabled. But, now that we are moving YouTube to use open graph data for everything, we could just replace the URL with the title always. You can make that change in a separate commit, if you prefer, since it changes behaviour for users.